packages/starlight/utils/route-data.ts

```diff
@@ -78,11 +77,10 @@ function getLastUpdated({ entry }: PageProps): Date | undefined {
 	const { lastUpdated: configLastUpdated } = config;
 
 	if (frontmatterLastUpdated ?? configLastUpdated) {
-		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + entry.id, project.root));
```

Comment:
Removing this changes the argument being passed down to `getNewestCommitDate()` to no longer being an absolute path.

This means that `getNewestCommitDate()` in dev would use `.` for the `cwd` option of `spawnSync` for a file like the getting started guide of the docs resulting in the following error:

> fatal: ambiguous argument 'getting-started.mdx': unknown revision or path not in the working tree.
> Use '--' to separate paths from revisions, like this:
> 'git <command> [<revision>...] -- [<file>...]'

And the last update date not being visible in dev mode.

A small reproduction is to just add `lastUpdated: true` to the `docs/astro.config.mjs` file.

Reply:
Receiving the absolute path would require inlining it in the SSR bundle, which exposes the project's path on the user machine.

I think it is better to resolve the relative path to the absolute path in the `getNewestCommitDate` when running the dev server.

Reply:
And actually this breaks for projects using a custom `srcDir`. So I'll have to fix that one together 😅